daemon/cmd/daemon.go

```diff
@@ -525,6 +525,8 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 		d.k8sWatcher.NodeChain.Register(&d.k8sWatcher.K8sSvcCache)
 	}
 
+	d.k8sWatcher.NodeChain.Register((*watchers.CiliumNodeLabelsUpdater)(d.k8sWatcher))
```

Comment:
`d.k8sWatcher.NodeChain.Register((*watchers.CiliumNodeLabelsUpdater)(d.k8sWatcher))` can be `d.k8sWatcher.NodeChain.Register(d.k8sWatcher)`. The code will compile as well.